### ISSUE TITLE
sclang executable cannot be found

### ISSUE THREAD

**Issue by pstahlhofen** [interactive-sonification/sc3nb#14]
**Problem Description**

After installing `sc3nb` as described in the `README.md` by downloading the source code and installing the package with `pip -e`, I started a Python interpreter in my terminal and ran the following two commands:
```
import sc3nb as scn

sc = scn.startup()
```
which triggered this error message:
```
Traceback (most recent call last):                                                │~                                                                                 
  File "<stdin>", line 1, in <module>                                             │~                                                                                 
  File "/Users/paulstahlhofen/Documents/Auditory_Data_Science/sc3nb/sc3nb/sc.py", │~                                                                                 
line 640, in startup                                                              │~                                                                                 
    sc = SC(**kwargs)                                                             │~                                                                                 
  File "/Users/paulstahlhofen/Documents/Auditory_Data_Science/sc3nb/sc3nb/sc.py", │~                                                                                 
line 81, in __init__                                                              │~                                                                                 
    sclangpath = find_executable('sclang', path=sclangpath)                       │~                                                                                 
  File "/Users/paulstahlhofen/Documents/Auditory_Data_Science/sc3nb/sc3nb/tools.py│~                                                                                 
", line 227, in find_executable                                                   │~                                                                                 
    raise FileNotFoundError("Unable to find sclang executable")                   │~                                                                                 
FileNotFoundError: Unable to find sclang executable
```

**Quick Fix**
After searching the `sclang` executable for a while, I was able to locate it in my installation of SuperCollider. If you encountered the same problem, I would recommend using `find` in your terminal as follows
```
find /path/to/SuperCollider -name 'sclang' -print
```
On my system, the output was
```
/Applications/SuperCollider/SuperCollider.app/Contents/MacOS/sclang
```
Now, you can add this to your path variable using
```
sclangpath='/Applications/SuperCollider/SuperCollider.app/Contents/MacOS:'; export PATH=$sclangpath$PATH
```
where my example path should be replaced by the path on your system and the trailing `/sclang` must be substituted by a colon
**Note**: copying the `sclang` executable to some directory already contained in your `$PATH` does NOT work (at least not in my case). I got problems with dynamic loading of libraries

**Request to the Developers**
Could you please document this problem and the quick fix briefly in your `README.md` in case other users encounter it? Also, implementing a prompt in `sc3nb` that asks for the `sclang` path in case the executable cannot be found would be a nice feature. Something like
```
sclang executable not found. Please enter the path to sclang (Hint: the executable will most likely be located in your installation of SuperCollider) >>>
```
Then the `sclang` path could be stored somewhere internally by `sc3nb` rather than requiring the user to change the system's path variable in order to get this package working.
Thank you!

**Comment by dreinsch** [comment]
Thank you for reporting this issue and your feedback!

We are currently preparing a new and improved version of sc3nb. There we provided more information in the documentation.
See (https://interactive-sonification.github.io/sc3nb/refactoring/autogen/notebooks/sc3nb-examples.html#Starting-sc3nb)

**Proving the executable paths to sc3nb**


- **via keyword argument**
As noted in the documentation you can also instead of changing your `PATH`  provide the path with the executables with `sclangpath` (for the current master version, `sclang_path` in the upcoming releases > 0.1) as a keyword argument to `startup`.

- **via PATH variable**
Also be aware that the purpose of the `PATH` variable (https://en.wikipedia.org/wiki/PATH_(variable)) is to list the locations of the executables.  However I agree that it can be tiring to add executables to the `PATH`. 
I would recommend you use the PATH separator between the variables.
`export PATH=$PATH:$sclangpath`
Have a look at https://unix.stackexchange.com/questions/26047/how-to-correctly-add-a-path-to-path for some best practices for extending the PATH.

**Next steps / some ideas**

- What do you think about the information from the documentation does it cover all the details you wanted?
- I like the idea of the interactive startup but for this we need to ensure the session is interactive, otherwise the startup would end up waiting for user input. 
- Another idea would be to extend the search for executables by some default installation directories. Then we could simply search in these paths aswell. However as you might have noticed our macOS path is already slightly different from yours. You have an additional SuperCollider folder before the SuperCollider.app folder.

**Comment by dreinsch** [comment]
I have added default installations paths to the search with the commit https://github.com/interactive-sonification/sc3nb/pull/13/commits/3a3d66db8bde9a5536478b5c17b94afdc6ef413f in #13.

This should fix this issue when #13 is merged.

**Comment by pstahlhofen** [comment]
Hello Dennis,
thank you! I just had a look at the code and I agree that it should solve the problem for my use case. Sorry for non-responding to your first comment. I had a busy time and it slipped my mind.
